### PR TITLE
SDI-537 Happy path for update adjustment

### PR DESCRIPTION
**Pull request by andymarke**
TODO:

- Exception paths
- Mapping doesn't exist path

### CODE REVIEW

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt [hunk 1]
```diff
@@ -680,6 +818,20 @@ class SentencingAdjustmentsToNomisTest : SqsIntegrationTestBase() {
     ).get()
   }
 
+  private fun publishUpdateAdjustmentDomainEvent() {
+    val eventType = "sentencing.sentence.adjustment.updated"
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt:822]: is "sentence" correct here?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 2]
```diff
@@ -95,6 +96,17 @@ class NomisApiService(@Qualifier("nomisApiWebClient") private val webClient: Web
       .bodyToMono(CreateSentencingAdjustmentResponse::class.java)
       .awaitSingle()
 
+  suspend fun updateSentenceAdjustment(
+    adjustmentId: Long,
+    request: UpdateSentencingAdjustmentRequest
+  ): Unit? =
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:102]: `Unit?` looks wrong

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt (hunk 1 above)
Comment (andymarke) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsToNomisTest.kt:822]: Probably not. But all made up right now. No idea what real event will be

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:102]: Yep. Was due to to awaitFirst which didn't work (there is no body), so did awaitFirstOrNull which forced a nullable Unit

I'll take a second look

**Pull request merged**